test/extensions/filters/http/transformation/inja_transformer_replace_test.cc

```diff
@@ -0,0 +1,491 @@
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::REPLACE_ALL);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("bar bar bar");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+
+  EXPECT_THROW(std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc)), EnvoyException);
+}
+
+TEST(Extraction, ReplaceAllHappyPath) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}, {"foo", "bar"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex("bar");
+  extractor.set_subgroup(0);
+  auto replacement_text = "BAZ";
+  extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::REPLACE_ALL);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("bar bar bar");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("BAZ BAZ BAZ", res);
+}
+
+TEST(Extraction, IndividualReplaceIdentity) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}, {"foo", "bar"}};
+
+  // Note that the regex contains enough capture groups
+  // that this (in theory) could be valid subgroup
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex(".*(bar).*");
+  extractor.set_subgroup(1);
+  auto replacement_text = "bar";
+  extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("bar bar bar");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("bar bar bar", res);
+}
+
+TEST(Extraction, ReplaceAllIdentity) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}, {"foo", "bar"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex("bar");
+  extractor.set_subgroup(0);
+  auto replacement_text = "bar";
+  extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::REPLACE_ALL);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("bar bar bar");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("bar bar bar", res);
+}
+
+TEST(Extraction, ReplaceAllNoMatch) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}, {"foo", "bar"}};
+
+  // Note that the regex contains enough capture groups
```

Comment:
does this comment belong here?

Reply:
Resolved, w/ other test cleanup here: 202d18dfc7ca07ede7f41ed2f5abefcb9a2d0126